airflow/models/baseoperator.py

```diff
@@ -127,7 +127,9 @@
 
 
 def parse_retries(retries: Any) -> int | None:
-    if retries is None or type(retries) == int:  # noqa: E721
+    if retries is None:
```

Comment:
I think here you could just do `return retries or 0` instead of `if/elif`. And then the change in airflow/models/taskinstance.py becomes unnecessary.

Reply:
I believe this will not work as the input might be `Any` and then the promise to return is is not met.

But otherwise I agree wit the change in baseoperator.py the change in taskinstance.py is not needed

Reply:
@PApostol @jscheffl Thank you for the review comments, I have removed the change in taskinstance.py since the change in baseoperator will handle the None values